src/ui/ui_settings/ui_settings_service.js

```diff
@@ -0,0 +1,130 @@
+import { defaultsDeep, noop } from 'lodash';
+import { errors as esErrors } from 'elasticsearch';
+
+import { getDefaultSettings } from './defaults';
+
+function hydrateUserSettings(userSettings) {
+  return Object.keys(userSettings)
+    .map(key => ({ key, userValue: userSettings[key] }))
+    .filter(({ userValue }) => userValue !== null)
+    .reduce((acc, { key, userValue }) => ({ ...acc, [key]: { userValue } }), {});
+}
+
+export class UiSettingsService {
+  constructor(options = {}) {
```

Comment:
++ I think we should stay away from optionals like this as much as possible, and if we want to define one it should list all the key/value pairs with their defaults.